packages/gatsby/src/redux/actions/public.js

```diff
@@ -447,6 +448,19 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
     )
   }
 
+  if (state.queries.createPagesFinished) {
+    reporter.warn(
+      chalk.bold.yellow(
+        `${name} attempted to create a page after the createPages API has finished.`
+      ) +
+        `Please check that the createPages implementation in gatsby-node is awaited properly.\n\nThe page object passed to createPage:\n${JSON.stringify(
+          page,
+          null,
+          2
+        )}`
+    )
```

Comment:
1. Can we potentially get following back, too?
```
const possiblyCodeFrame = getNonGatsbyCodeFrameFormatted()
            if (possiblyCodeFrame) {
              warning.push(possiblyCodeFrame)
            }
```

This was what was showing which of the `createPage` calls was "outside" (if the helper function can figure it out), which should be easier to track for users as the `page` we will print out will likely come from some dynamic (sourced) data.

2. Can we keep the warning deduplication? At scale this is potentially very spammy